### PR TITLE
Phoenix: from chapter 7 to 9

### PR DESCRIPTION
**Pull request by Vretamoza**

### CODE REVIEW

File: hello/lib/hello_web.ex [hunk 1]
```diff
@@ -35,7 +35,8 @@ defmodule HelloWeb do
 
       # Import convenience functions from controllers
       import Phoenix.Controller,
-        only: [get_flash: 1, get_flash: 2, view_module: 1, view_template: 1]
+        only: [get_flash: 1, get_flash: 2, view_module: 1,
```
Comment (juanhurtado10) [inline hello/lib/hello_web.ex:38]: @Vretamoza what is the benefit of using `only` when importing from a module?

File: hello/lib/hello_web/plugs/locale.ex [hunk 2]
```diff
@@ -0,0 +1,15 @@
+defmodule HelloWeb.Plugs.Locale do
+  import Plug.Conn
+
+  @locales ["en", "fr", "de"]
+
+  def init(default), do: default
+
+  def call(%Plug.Conn{params: %{"locale" => loc}} = conn, _default) when loc in @locales do
```
Comment (juanhurtado10) [inline hello/lib/hello_web/plugs/locale.ex:8]: @Vretamoza how does the call function work in a plug?

File: hello/lib/hello_web/templates/layout/admin.html.heex [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <link phx-track-static rel="stylesheet" href={Routes.static_path(@conn, "/assets/app.css")}/>
+    <script defer phx-track-static type="text/javascript" src={Routes.static_path(@conn, "/assets/app.js")}></script>
+  </head>
+  <body>
+    <header>
+      <section class="container">
+        <nav>
+          <ul>
+            <li><a href="https://hexdocs.pm/phoenix/overview.html">Get Started</a></li>
+            <%= if function_exported?(Routes, :live_dashboard_path, 2) do %>
```
Comment (juanhurtado10) [inline hello/lib/hello_web/templates/layout/admin.html.heex:18]: @Vretamoza could you please explain in deep `function_exported?/3`?

File: hello/lib/hello_web/templates/layout/root.html.heex [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
     <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
     <%= csrf_meta_tag() %>
-    <%= live_title_tag assigns[:page_title] || "Hello", suffix: " · Phoenix Framework" %>
```
Comment (juanhurtado10) [inline hello/lib/hello_web/templates/layout/root.html.heex:8]: @Vretamoza why is this change needed?

File: hello/lib/hello_web/views/layout_view.ex [hunk 5]
```diff
@@ -1,6 +1,10 @@
 defmodule HelloWeb.LayoutView do
   use HelloWeb, :view
 
+
+  def title() do
```
Comment (juanhurtado10) [inline hello/lib/hello_web/views/layout_view.ex]: @Vretamoza by coding style functions with arity: 0 should not have parentheses

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: hello/lib/hello_web.ex (hunk 1 above)
Comment (Vretamoza) [reply hello/lib/hello_web.ex:38]: The benefit is that we get just the functionalities that we need instead of bringing all of them.

File: hello/lib/hello_web/plugs/locale.ex (hunk 2 above)
Comment (Vretamoza) [reply hello/lib/hello_web/plugs/locale.ex:8]: It receives as second parameter what the init function returns and does what it is meant to do.

File: hello/lib/hello_web/templates/layout/admin.html.heex (hunk 3 above)
Comment (Vretamoza) [reply hello/lib/hello_web/templates/layout/admin.html.heex:18]: It receives three parameters, a module, a function and its arity. The function returns true if the given module is loaded and has a function with the given arity. If that's not the case, it reurns false.

File: hello/lib/hello_web/templates/layout/root.html.heex (hunk 4 above)
Comment (Vretamoza) [reply hello/lib/hello_web/templates/layout/root.html.heex:8]: Because we declared the function title/0 in the LayoutView, so we can just call it since the layout template is the respective template of the LayoutView.

File: hello/lib/hello_web/views/layout_view.ex (hunk 5 above)
Comment (Vretamoza) [reply hello/lib/hello_web/views/layout_view.ex]: I changed it already

File: hello/lib/hello_web/templates/layout/admin.html.heex (hunk 3 above)
Comment (juanhurtado10) [reply hello/lib/hello_web/templates/layout/admin.html.heex:18]: And why use it? what is the reason behind it?

File: hello/lib/hello_web/plugs/locale.ex (hunk 2 above)
Comment (juanhurtado10) [reply hello/lib/hello_web/plugs/locale.ex:8]: 🤔  could you please explain in deep?

File: hello/lib/hello_web/templates/layout/root.html.heex (hunk 4 above)
Comment (juanhurtado10) [reply hello/lib/hello_web/templates/layout/root.html.heex:8]: @Vretamoza and the `live_title_tag` didn't work for this purpose as well?

**Comment by juanhurtado10** [comment]
@Vretamoza new comments were added. Please, rebase your branch as well

File: hello/lib/hello_web/plugs/locale.ex (hunk 2 above)
Comment (Vretamoza) [reply hello/lib/hello_web/plugs/locale.ex:8]: Plugs take requests and modify them as needed, in the init callback we initialize the arguments and pass them as second parameter in the call function which is the one that makes the required changes of the plug.

File: hello/lib/hello_web/templates/layout/admin.html.heex (hunk 3 above)
Comment (Vretamoza) [reply hello/lib/hello_web/templates/layout/admin.html.heex:18]: It is used to check if the module and the function I'm going to use is loaded, but if it's not, it doesn't load it.

File: hello/lib/hello_web/templates/layout/root.html.heex (hunk 4 above)
Comment (Vretamoza) [reply hello/lib/hello_web/templates/layout/root.html.heex:8]: With the function title/0 we can define a custom title, unlike live_title_tag which uses the :page_title object to assign a title depending on the context.
